cloudconfig/cloudinit/cloudinit_ubuntu.go

```diff
@@ -300,13 +304,46 @@ func (cfg *ubuntuCloudConfig) addRequiredPackages() {
 }
 
 // Updates proxy settings used when rendering the conf as a script
-func (cfg *ubuntuCloudConfig) updateProxySettings(proxySettings proxy.Settings) {
+func (cfg *ubuntuCloudConfig) updateProxySettings(proxyCfg PackageManagerProxyConfig) error {
 	// Write out the apt proxy settings
-	if (proxySettings != proxy.Settings{}) {
+	if aptProxy := proxyCfg.AptProxy(); (aptProxy != proxy.Settings{}) {
+		pkgCmder := cfg.paccmder[jujupackaging.AptPackageManager]
 		filename := config.AptProxyConfigFile
 		cfg.AddBootCmd(fmt.Sprintf(
 			`printf '%%s\n' %s > %s`,
-			utils.ShQuote(cfg.paccmder.ProxyConfigContents(proxySettings)),
+			utils.ShQuote(pkgCmder.ProxyConfigContents(aptProxy)),
 			filename))
 	}
+
+	// Write out the snap http/https proxy settings
+	if snapProxy := proxyCfg.SnapProxy(); (snapProxy != proxy.Settings{}) {
```

Comment:
Not really a fan of this `(snapProxy != proxy.Settings{})` - I guess a small tiny comment about checking if the `snapProxy` is not empty.

Reply:
Yeah, it was already there so I decided to keep it. Fun fact: if you remove the parentheses it does not compile!